viam-cartographer_internal_test.go

```diff
@@ -442,50 +442,101 @@ func setMockGetPointCloudFunc(
+	api func(context.Context) (func() ([]byte, error), error),
+	fileIsLargerThanChunkSize bool,
+) {
+	bytes, err := os.ReadFile(artifact.MustPath(path))
+	test.That(t, err, test.ShouldBeNil)
+
+	if fileIsLargerThanChunkSize {
+		test.That(t, len(bytes), test.ShouldBeGreaterThan, 1024*1024)
+	} else {
+		test.That(t, len(bytes), test.ShouldBeLessThan, 1024*1024)
+	}
+
+	setMockFunc(mockCartoFacade, bytes)
+
+	getAndCheckCallbackFunc(t, api, false)
+}
+
+func getAndCheckCallbackFunc(
+	t *testing.T,
+	api func(context.Context) (func() ([]byte, error), error),
+	emptyBytes bool,
+) {
+	callback, err := api(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	bytes, err := slam.HelperConcatenateChunksToFull(callback)
+	test.That(t, err, test.ShouldBeNil)
+
+	if emptyBytes {
+		test.That(t, bytes, test.ShouldBeNil)
+	} else {
+		test.That(t, bytes, test.ShouldNotBeNil)
+	}
+}
 
+func testApisThatReturnCallbackFuncs(
+	t *testing.T,
+	svc *cartographerService,
+	mockCartoFacade *cartofacade.Mock,
+	pathToFileLargerThanChunkSize string,
+	pathToFileSmallerThanChunkSize string,
+	api func(context.Context) (func() ([]byte, error), error),
+	setMockFunc func(*cartofacade.Mock, []byte),
+) {
 	svc.cartofacade = mockCartoFacade
 	svc.modularizationV2Enabled = true
 
-	t.Run("pointcloud smaller than 1 mb limit - success", func(t *testing.T) {
-		file := "viam-cartographer/outputs/viam-office-02-22-3/pointcloud/pointcloud_0.pcd"
-		inputPointCloudMapBytes, err := os.ReadFile(artifact.MustPath(file))
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, len(inputPointCloudMapBytes), test.ShouldBeLessThan, 1024*1024)
-
-		setMockGetPointCloudFunc(mockCartoFacade, inputPointCloudMapBytes)
-		callback, err := svc.GetPointCloudMap(context.Background())
-		test.That(t, err, test.ShouldBeNil)
-		pointCloudMapBytes, err := slam.HelperConcatenateChunksToFull(callback)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, pointCloudMapBytes, test.ShouldResemble, inputPointCloudMapBytes)
+	t.Run("returned value smaller than 1 mb limit - success", func(t *testing.T) {
+		path := pathToFileLargerThanChunkSize
+		testApisThatReturnCallbackFuncsSuccess(
+			t,
+			path,
+			setMockFunc,
+			mockCartoFacade,
+			api,
+			false,
+		)
 	})
 
-	t.Run("pointcloud larger than 1 mb limit - success", func(t *testing.T) {
-		file := "viam-cartographer/outputs/viam-office-02-22-3/pointcloud/pointcloud_1.pcd"
-		inputPointCloudMapBytes, err := os.ReadFile(artifact.MustPath(file))
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, len(inputPointCloudMapBytes), test.ShouldBeGreaterThan, 1024*1024)
-
-		setMockGetPointCloudFunc(mockCartoFacade, inputPointCloudMapBytes)
-		callback, err := svc.GetPointCloudMap(context.Background())
-		test.That(t, err, test.ShouldBeNil)
-		pointCloudMapBytes, err := slam.HelperConcatenateChunksToFull(callback)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, pointCloudMapBytes, test.ShouldResemble, inputPointCloudMapBytes)
```

Comment:
Where are we actually asserting that both the output of the GetPointCloudMap & GetInternalState call back functions, when concatenated together, both above & below the chunk limit, equal the files they were derived from?

Reply:
i must've accidentally removed that during the refactor fixed and happening [here](https://github.com/viamrobotics/viam-cartographer/blob/88baa8b90fea27333d91f60991b0d572bc26d522/viam-cartographer_internal_test.go#L481)